sdk/eventhub/event-hubs/src/partitionLoadBalancer.ts

```diff
@@ -129,18 +129,26 @@ export class FairPartitionLoadBalancer implements PartitionLoadBalancer {
     partitionIds: string[],
     ownerPartitionMap: Map<string, PartitionOwnership[]>
   ): boolean {
-    const numberOfPartitionsOwned = ownerPartitionMap.get(ourOwnerId)!.length;
+    const numPartitionsOwnedByUs = ownerPartitionMap.get(ourOwnerId)!.length;
+
+    if (numPartitionsOwnedByUs < minPartitionsPerEventProcessor) {
+      return true;
+    }
 
     let sumOfPartitionsOwnedByAnyProcessor = 0;
+
     for (const ownershipList of ownerPartitionMap.values()) {
       sumOfPartitionsOwnedByAnyProcessor =
         sumOfPartitionsOwnedByAnyProcessor + ownershipList.length;
     }
-    return (
-      numberOfPartitionsOwned < minPartitionsPerEventProcessor ||
-      (sumOfPartitionsOwnedByAnyProcessor < partitionIds.length &&
-        numberOfPartitionsOwned < minPartitionsPerEventProcessor + 1)
-    );
+
+    const thereAreUnownedPartitions = sumOfPartitionsOwnedByAnyProcessor < partitionIds.length;
+
+    if (thereAreUnownedPartitions && numPartitionsOwnedByUs < minPartitionsPerEventProcessor) {
+      return true;
+    }
+
+    return false;
```

Comment:
Can this whole function just be changed to:
```typescript
const numPartitionsOwnedByUs = ownerPartitionMap.get(ourOwnerId)!.length;
return (numPartitionsOwnedByUs < minPartitionsPerEventProcessor);
```
The check you're doing on line 147 is already handled by the check on line 134, and since `thereAreUnownedPartitions` isn't used anywhere outside this check, I don't think it's even needed.

Reply:
And if we make this change, maybe also change `minPartitionsPerEventProcessor` to `targetPartitionCount` or something like that, since we're adding the extra partition to this param before calling this method.

Reply:
I like both of those ideas. :)

 I just dumbly expanded the existing function but it's even more obvious now that the logic is kind of silly :)

Reply:
The more I looked at it the more it bugged me. I modified the code to be just use some of the counts that the _isLoadBalanced() method was already doing which made things (hopefully) clearer. 

That also involved making this a simpler check as well.